### PR TITLE
Fix AuthenticationDialog crash; Add support for testing Authenticatio…

### LINKED ISSUE #1234: Dialog version of AuthenticationContext.acquireToken() causes subsequent non-ADAL Chromium webviews to throw
### Steps:
On a device that uses Chromium for its WebViews by default:

1. Call AuthenticationContext.acquireToken() ([a Dialog-based one](http://static.javadoc.io/com.microsoft.aad/adal/1.14.0/com/microsoft/aad/adal/AuthenticationContext.html#acquireToken-java.lang.String-java.lang.String-java.lang.String-java.lang.String-com.microsoft.aad.adal.PromptBehavior-java.lang.String-com.microsoft.aad.adal.AuthenticationCallback-))
2. On the UI thread, call anything that causes a WebView to inflate (ie: create another Dialog, setContentView(R.layout.some_layout_with_webview)

### Expected:
The new WebView inflates successfully.

### Actual:
An IllegalStateException is thrown:

> E/AndroidRuntime: FATAL EXCEPTION: AcquireTokenRequestHandlerThread
> java.lang.IllegalStateException: Calling View methods on another thread than the UI thread.
> at com.android.webview.chromium.WebViewChromium.createThreadException(WebViewChromium.java:294)
> at com.android.webview.chromium.WebViewChromium.checkThread(WebViewChromium.java:309)
> at com.android.webview.chromium.WebViewChromium.init(WebViewChromium.java:227)
> at android.webkit.WebView.<init>(WebView.java:606)
> at android.webkit.WebView.<init>(WebView.java:542)
> at android.webkit.WebView.<init>(WebView.java:525)
> at android.webkit.WebView.<init>(WebView.java:512)
> at java.lang.reflect.Constructor.newInstance(Native Method)
> at android.view.LayoutInflater.createView(LayoutInflater.java:619)
> at com.android.internal.policy.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:58)
> at android.view.LayoutInflater.onCreateView(LayoutInflater.java:694)
> at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:762)
> at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:704)
> at android.view.LayoutInflater.rInflate(LayoutInflater.java:835)
> at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:798)
> at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
> at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
> at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
> at com.android.internal.policy.PhoneWindow.setContentView(PhoneWindow.java:393)
> at android.app.Dialog.setContentView(Dialog.java:512)

This is on a device running Android 6.0.1, ADAL 1.14.0.

### Speculative cause:
In my logs, when I run acquireToken(), I see the following a few lines after:
> V/WebViewChromiumFactoryProvider: Binding Chromium to background looper Looper (AcquireTokenRequestHandlerThread, tid 2626) {b98db11}

Chromium web views appear to [bind to the thread that they are first called on](https://android.googlesource.com/platform/frameworks/webview/+/f989a7846d5fe4116de1a0635a8aade09d2be7b2/chromium/java/com/android/webview/chromium/WebViewChromiumFactoryProvider.java#199), treating it as the UI thread. They will then [throw if called from any other thread](https://android.googlesource.com/platform/frameworks/webview/+/f989a7846d5fe4116de1a0635a8aade09d2be7b2/chromium/java/com/android/webview/chromium/WebViewChromium.java#354).

Since #906, ADAL has been calling its webview methods on a private thread rather than the UI thread. Chromium binds to this, making webviews inoperable from any other thread. I can successfully work around this issue by rolling back to ADAL version 1.12.0, before this change was added.

Assuming the above cause is correct, is it possible to fix the memory leak mentioned in #906 some other way?

### PR DESCRIPTION
**Pull request by shoatman**
…nDialog to automation app; Add stress test to automation test app; Add web view open to automation test app

This closes #1234

### CODE REVIEW

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/CustomWebViewClient.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.microsoft.aad.automation.testapp;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+public class CustomWebViewClient extends WebViewClient {
```
Comment (kreedula) [inline automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/CustomWebViewClient.java]: Just curious why a CustomWebViewClient if we are just returning false from shouldOverrideUrl

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/MainActivity.java [hunk 2]
```diff
@@ -128,6 +145,12 @@ private void launchAuthenticationInfoActivity(int flowCode) {
         this.startActivity(intent);
     }
 
+    private void openWevViewActivity(){
```
Comment (kreedula) [inline automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/MainActivity.java]: Typo : openWebViewActivity

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/MainActivity.java [hunk 3]
```diff
@@ -128,6 +145,12 @@ private void launchAuthenticationInfoActivity(int flowCode) {
...
+        final Intent intent = new Intent();
+        intent.setClass(mContext, WebViewActivity.class);
```
Comment (kreedula) [inline automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/MainActivity.java]: Nit : we can use Intent's constructor to set class and context directly like

 `final Intent intent = new Intent(mContext, WebViewActivity.class);`

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/SignInActivity.java [hunk 4]
```diff
@@ -148,9 +156,19 @@ private void performAuthentication() {
         validateUserInput(inputItems, flowCode);
 
         setAuthenticationData(inputItems);
-        AuthenticationSettings.INSTANCE.setUseBroker(true);
+        AuthenticationSettings.INSTANCE.setUseBroker(mUseBroker);
```
Comment (kreedula) [inline automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/SignInActivity.java]: I had the same change which went in. Conflict needs to be resolved here

**Comment by iambmelt** [comment]
Nothing further to add to @kreedula at this time

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/CustomWebViewClient.java (hunk 1 above)
Comment (shoatman) [reply automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/CustomWebViewClient.java]: False indicates that the webview itself handles the redirect.  Which is fine in this case.

File: automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/MainActivity.java (hunk 3 above)
Comment (shoatman) [reply automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/MainActivity.java]: I'll fix since I'm fixing the typo

**Comment by kreedula** [comment]
Thanks @shoatman. LGTM 👍

**Comment by iambmelt** [comment]
Linking to related items.

Original issue: #908 
Original PR: #948 

Related: #888, #906 

Alternative 'fix' proposed by @iambmelt: #1094

**Comment by iambmelt** [comment]
Related: https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1261

**Pull request merged**